YourEasyRentTest/Controllers/ProductControllerTest.cs

```diff
@@ -0,0 +1,264 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using FluentAssertions;
+using YourEasyRent.Entities;
+using YourEasyRent.DataBase.Interfaces;
+using YourEasyRent.Controllers;
+using YourEasyRent.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace YourEasyRentTest.Controllers
+{
+    public class ProductControllerTest
+    {
+        [Fact]
+        public async Task GetProducts_Return_AllProducts()
+        {
+            // arrange
+            var products = new List<Product>{new Product { Id = "test", Name = "Product 1" }, new Product { Id = "test", Name = "Product 2" } };       
+            var productRepoMock = new Mock<IProductRepository>(); 
+            productRepoMock.Setup(repo => repo.GetProducts()).ReturnsAsync(products);
+            var controller = new ProductController(productRepoMock.Object); // Создала экземпляр контроллера ProductsController, передавая фейковый repositoryMock в конструктор.
+
+            // act
+            var result = await controller.GetProducts();
+
+            // assert
+            result.Should().BeOfType<ActionResult<IEnumerable<Product>>>()
+                .Which.Value.Should().BeAssignableTo<IEnumerable<Product>>()
+                .And.HaveCount(products.Count); // проверряем является ли результат объектом типа <ActionResult<IEnumerable<Product>, в котором значение Value является IEnumerable<Product> и проверяем что количество элементов в IEnumerable<Product> равно количеству продуктов, которые мы заранее подготовили.
+
+        }
+
+        [Fact]
+        public async Task GetProducts_Return_ProductsNotFound()
```

Comment:
Name should be more descriptive. And tell us when and what it does. In this case it's
GetProducts_WhenNoProductsFound_ShouldReturnNotFound()